advanced_alchemy/repository/_async.py

```diff
@@ -148,12 +172,34 @@ def check_not_found(item_or_none: ModelT | None) -> ModelT:
             raise NotFoundError(msg)
         return item_or_none
 
+    def _reset_loader_options(self) -> None:
+        self._loader_options, self._loader_options_have_wildcards = get_abstract_loader_options(
+            loader_options=self._default_loader_options,
+            default_options_have_wildcards=self._loader_options_have_wildcards,
+        )
+
+    def _set_loader_options(self, loader_options: LoadSpec | None) -> None:
+        self._loader_options, self._loader_options_have_wildcards = get_abstract_loader_options(
+            loader_options=loader_options,
+            default_loader_options=self._default_loader_options,
+            default_options_have_wildcards=self._loader_options_have_wildcards,
+        )
+
+    def _reset_execution_options(self) -> None:
+        self._execution_options = self._default_execution_options
+
+    def _set_execution_options(self, execution_options: dict[str, Any] | None = None) -> None:
+        if execution_options is not None:
+            self._execution_options = execution_options
+
```

Comment:
This API, or rather its usage, is partially broken. For async usage it's fine as it's being used currently, but the sync one is not thread safe. Because this relies on attributes being mutated on an instance, and these mutations aren't guarded in any way, when this is being done in 2 threads, thread 1 might set the options, hand of to thread 2, which overwrites the options with its own and hand back the control, at which point thread 1 attempts to perform the operations with the options set by thread 2.

The same scenario could happen with async context switching if multiple coroutines make use of the options within a single function call after having called e.g. `_set_loader_options`.

This could either be fixed by using contextvars or, preferably, by simply passing in the options to the methods that might want to use them.